test/alternator/conftest.py

```diff
@@ -89,6 +82,21 @@ def dynamodb(request):
             region_name='us-east-1', aws_access_key_id='alternator', aws_secret_access_key='secret_pass',
             config=boto_config.merge(botocore.client.Config(retries={"max_attempts": 0}, read_timeout=300)))
 
+# "dynamodb" fixture: set up client object for communicating with the DynamoDB
+# API. Currently this chooses either Amazon's DynamoDB in the default region
+# or a local Alternator installation on http://localhost:8080 - depending on the
+# existence of the "--aws" option. In the future we should provide options
+# for choosing other Amazon regions or local installations.
+# We use scope="session" so that all tests will reuse the same client object.
+@pytest.fixture(scope="session")
+def dynamodb(request):
+    return _dynamodb(request)
+
+# "dynamodb" fixture for tests which want their own client instance.
+@pytest.fixture(scope="function")
+def dynamodb_function_scoped(request):
+    return _dynamodb(request)
```

Comment:
I'm worried that the fact that these fixtures all "return" the resource instead of yield+cleanup, and that the new _dynamodb is a function and not a contextmanager, was a mistake on my part when I created the dynamodb fixture. Shouldn't we need to close() the resource? I'm worried that it didn't matter when we had just one "dynamodb" fixture for the entire run, but will start mattering when we have all those temporary sessions.

Reply:
ok, switched to non fixture func